railties/lib/rails/generators/rails/authentication/templates/test/helpers/session_test_helper.rb.tt

```diff
@@ -0,0 +1,15 @@
+module SessionTestHelper
+  def sign_in_as(user)
+    Current.session = user.sessions.create!
+
+    ActionDispatch::TestRequest.create.cookie_jar.tap do |cookie_jar|
+      cookie_jar.signed[:session_id] = Current.session.id
+      cookies[:session_id] = cookie_jar[:session_id]
```

Comment:
Is this supposed to be added o system tests or request tests? I don't remember if system tests can access cookies

Reply:
This is for IntegrationTests (ActionDispatch::IntegrationTest).

Reply:
Should we have one for system tests too? I know @dhh is not a big fan of them, but for smoke tests, a set of system tests are useful. Should we let that for each app to implement? Maybe a initial implementation on the docs could be an alternative, if we don't want to ship it in the framework.

Reply:
Well good question. I think system tests are probably way simpler. These helpers here were not trivial because of the cookie jar problem mentioned in the referenced discussion.

For system tests it's pretty straightforward I guess. Something like:

```
visit new_session_url
fill_in "Email", with: "test@example.com"
fill_in "Password", with: "password123"
click_on "Sign in"
```

Where should logic for this go? Directly in the ApplicationSystemTestCase class?
Anyways I needed the IntegrationTest functionality as I was repeating it in all my projects. I let you guys decide how to move forward.

Reply:
I wonder if the system specs should actually login, or set cookies more "manually" too.

I'll defer to @dhh to say if this is necessary or not.

Reply:
For system tests, I needed to add in the following to get that solution to work:
```
page.instance_variable_set(:@touched, false)
page.reset!
```
However, this is pretty hacky, and I'm pretty sure there's a better solution for system tests

Reply:
Just an update, the solution in https://github.com/rails/rails/issues/53207#issuecomment-2628448719 works better for me than the page resetting! I think there was a significant amount of flake introduced by using stock capybara fill_in to initiate the session

Reply:
If it is for `ActionDispatch::IntegrationTest` why is is being added to `ActiveSupport::TestCase`